tools/functional-tester/etcd-agent/agent.go

```diff
@@ -80,8 +81,23 @@ func (a *Agent) stop() error {
 	if a.state != stateStarted {
 		return nil
 	}
-	err := a.cmd.Process.Kill()
-	if err != nil {
+
+	if err := syscall.Kill(a.cmd.Process.Pid, syscall.SIGTERM); err != nil {
+		return err
+	}
+	time.Sleep(time.Second)
+
+	if err := syscall.Kill(a.cmd.Process.Pid, syscall.SIGINT); err != nil {
+		return err
+	}
+	time.Sleep(time.Second)
+
+	if err := syscall.Kill(a.cmd.Process.Pid, syscall.SIGHUP); err != nil {
+		return err
+	}
+	time.Sleep(time.Second)
+
+	if err := a.cmd.Process.Kill(); err != nil {
```

Comment:
@heyitsanthony Just found out Wait returns ProcessState to check if the process exited or not. Will retry your way with it. Thanks.